lib/src/message.dart

```diff
@@ -61,12 +60,27 @@ abstract class Message {
       {Encoding encoding,
       Map<String, String> headers,
       Map<String, Object> context})
-      : this._(new Body(body, encoding), headers, context);
+      : this._forwardToContentType(body, encoding,
+            _determineMediaType(body, encoding, headers), headers, context);
 
-  Message._(Body body, Map<String, String> headers, Map<String, Object> context)
+  Message.withContentType(Body body, MediaType contentType,
```

Comment:
I don't want this constructor here. The shape of the constructors from before this CL is generally correct; the only change should be adding a `{bool bodyIsMap: false}` parameter to `new Message._()` and `_adjustHeaders()`.

Reply:
I had planned to use it with `http.MultiPart` since the `MediaType` requires a boundary which would be randomly generated.

Reply:
I'd rather have `_adjustHeaders()` do some logic like

```dart
if (body is MultipartBody) {
  contentType = "multipart/mixed; boundary=${body.boundary}";
}
```